src/rp2_common/hardware_rosc/rosc.c

```diff
@@ -30,8 +30,13 @@ uint rosc_find_freq(uint32_t low_mhz, uint32_t high_mhz) {
 }
 
 void rosc_set_div(uint32_t div) {
-    assert(div <= 31 && div >= 1);
-    rosc_write(&rosc_hw->div, ROSC_DIV_VALUE_PASS + div);
+    assert(div <= 31);
+    rosc_clear_bad_write();
+    assert(rosc_write_okay());
```

Comment:
Well that's a good question - it's so long since I wrote this PR that I can't remember now! I'll have to dig through my emails this evening....... (and hope that the answer exists in an email and isn't lost somewhere in the google chat history)